_posts/2018-02-27-optimizing-grpc-part-1.md

```diff
@@ -0,0 +1,395 @@
+---
+layout: post
+title: Optimizing gRPC - Part 1
+published: false
+permalink: blog/optimizing-grpc-part-1
+author: Carl Mastrangelo
+company: Google
+company-link: https://www.google.com
+---
+
+# Optimizing gRPC - Part 1
+
+A common question with gRPC is how to make it fast.  The gRPC library offers users access to high
+performance RPCs, but it isn't always clear how to achieve this.  Because this question is common 
+enough I thought I would try to show my thought process when tuning programs.
+
+## Setup
+
+Consider a basic Key Value service that is used by multiple other programs.  The service needs to
+be safe for concurrent access in case updates happen at the same time.  It needs to be able to scale
+up to use the available hardware.   Lastly, it needs to be fast.  gRPC is a perfect fit for this
+kind of service, so let's use it as our starting point.
```

Comment:
This sentence is a bit odd - we're writing a blog post about optimizing a gRPC service, so why would we choose anything else?  How about just "gRPC is a perfect fit for this type of service; now let's look at the best way to implement it."

Reply:
Done.